forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::{string::ToString, time::Instant};
+
+use clap::Subcommand;
+use colored::*;
+use forest_blocks::tipset_keys_json::TipsetKeysJson;
+use forest_cli_shared::cli::cli_error_and_die;
+use forest_rpc_client::{
+    chain_get_name, chain_get_tipset, chain_head, state_start_time, wallet_default_address,
+};
+
+use super::Config;
+use crate::cli::handle_rpc_err;
+
+#[derive(Debug, Subcommand)]
+pub enum InfoCommand {
+    Show,
+}
+
+#[derive(Debug, strum_macros::Display)]
+enum SyncStatus {
+    Ok,
+    Slow,
+    Behind,
+}
+
+pub struct NodeStatusInfo {
+    behind: u64,
+    health: usize,
+    epoch: i64,
+    base_fee: String,
+    sync_status: SyncStatus,
+}
+
+pub async fn node_status(config: &Config) -> NodeStatusInfo {
+    let chain_head = match chain_head(&config.client.rpc_token).await {
+        Ok(head) => head.0,
+        Err(_) => cli_error_and_die("Could not get network head. Is the node running?", 1),
+    };
+
+    let chain_finality = config.chain.policy.chain_finality;
+    let epoch = chain_head.epoch();
+    let ts = chain_head.min_timestamp();
+    let now = Instant::now().elapsed().as_secs();
+    let delta = ts - now;
+    let behind = delta / 30;
+
+    let sync_status = if delta < 30 * 3 / 2 {
+        SyncStatus::Ok
+    } else if delta < 30 * 5 {
+        SyncStatus::Slow
+    } else {
+        SyncStatus::Behind
+    };
+
+    let base_fee = chain_head.min_ticket_block().parent_base_fee();
+    let base_fee = base_fee.to_string();
+
+    // chain health
+    let blocks_per_tipset_last_finality = if epoch > chain_finality {
+        let mut block_count = 0;
+        let mut ts = chain_head;
+
+        for _ in 0..100 {
+            block_count += ts.blocks().len();
+            let tsk = ts.parents();
+            let tsk = TipsetKeysJson(tsk.clone());
+            if let Ok(tsjson) = chain_get_tipset((tsk,), &config.client.rpc_token).await {
+                ts = tsjson.0;
+            }
+        }
+
+        for _ in 100..chain_finality {
+            block_count += ts.blocks().len();
+            let tsk = ts.parents();
+            let tsk = TipsetKeysJson(tsk.clone());
+            if let Ok(tsjson) = chain_get_tipset((tsk,), &config.client.rpc_token).await {
+                ts = tsjson.0;
+            }
+        }
```

Comment:
What's the difference between these two `for` loops?

Reply:
This was simply a side effect of verbatim port,  except this one doesn't have the intermediary variable being set from lotus codebase:
```
status.ChainStatus.BlocksPerTipsetLast100 = float64(blockCnt) / 100
```
Didn't need that. 
I will combine them